### PR TITLE
Update authentication and authorization docs

### PR DESCRIPTION
**Pull request by indikasampath2000**
## Purpose
Modified authentication and authorization documentation explaining more details about customization.

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: docs/dev/security-architecture.md [hunk 1]
```diff
@@ -20,4 +20,58 @@ password, it will response authentication result with authentication Id.
...
   - [Changing the Default Security Settings](../user/authentication-settings-guide.md)
 
-If authentication is success, connection will be established or else will send authentication error.
\ No newline at end of file
+If authentication is success, connection will be established or else will send authentication error.
+
+### Writing a custom authenticator
+
+Custom authenticator should implement the [Authenticator](https://github.com/ballerina-platform/ballerina-message-broker/blob/master/modules/broker-auth/src/main/java/io/ballerina/messaging/broker/auth/authentication/Authenticator.java) interface and handle the 
+authentication underlying logic. Message broker only interested in [AuthResult](https://github.com/ballerina-platform/ballerina-message-broker/blob/master/modules/broker-auth/src/main/java/io/ballerina/messaging/broker/auth/authentication/AuthResult.java)
```
Comment (a5anka) [inline docs/dev/security-architecture.md]: `Message broker only interested ` -> `Message broker is only interested` ?

File: docs/dev/security-architecture.md [hunk 2]
```diff
@@ -20,4 +20,58 @@ password, it will response authentication result with authentication Id.
...
+2. Authenticate a user and provide a result to the message broker.
+
+    ```
+        AuthResult authenticate(String username, char[] password) 
+                        throws AuthException;
+    ```
+
+The default authenticator [JaasAuthenticator](https://github.com/ballerina-platform/ballerina-message-broker/blob/master/modules/broker-auth/src/main/java/io/ballerina/messaging/broker/auth/authentication/authenticator/JaasAuthenticator.java) 
+based on JAAS and provide custom LoginModule implementation call [UserStoreLoginModule](https://github.com/ballerina-platform/ballerina-message-broker/blob/master/modules/broker-auth/src/main/java/io/ballerina/messaging/broker/auth/authentication/jaas/UserStoreLoginModule.java).
+
```
Comment (a5anka) [inline docs/dev/security-architecture.md]: Shall we fix this as follows?
```
The default authenticator JaasAuthenticator is based on JAAS and provides a LoginModule implementation called UserStoreLoginModule.
```

File: docs/dev/security-architecture.md [hunk 3]
```diff
@@ -20,4 +20,58 @@ password, it will response authentication result with authentication Id.
  Users can defined their own JAAS Login module or write own custom authenticators as well.
   - [Changing the Default Security Settings](../user/authentication-settings-guide.md)
 
-If authentication is success, connection will be established or else will send authentication error.
\ No newline at end of file
+If authentication is success, connection will be established or else will send authentication error.
```
Comment (MaryamZi) [inline docs/dev/security-architecture.md]: shall we say "If authentication is successful"?

File: docs/user/authorization-settings-guide.md [hunk 4]
```diff
@@ -1,7 +1,10 @@
 # Changing the Default Authorization
 
-Mandatory Access Controller (MAC) and Discretionary Access Controller (DAC) use for the authorization in the Message
-Broker and it is disabled by default. User could enable it and restrict the resource access. 
+Mandatory Access Control (MAC) and Discretionary Access Control (DAC) use for the authorization in the Message
+Broker and it disabled by default. A user could enable it and restrict the resource access. After enabling,
```
Comment (a5anka) [inline docs/user/authorization-settings-guide.md]: it disabled -> is disabled

File: docs/dev/security-architecture.md (hunk 1 above)
Comment (indikasampath2000) [reply docs/dev/security-architecture.md]: Fixed in 41f8861

File: docs/dev/security-architecture.md (hunk 2 above)
Comment (indikasampath2000) [reply docs/dev/security-architecture.md]: Fixed in 41f8861

File: docs/dev/security-architecture.md (hunk 3 above)
Comment (indikasampath2000) [reply docs/dev/security-architecture.md]: Fixed in 41f8861

File: docs/user/authorization-settings-guide.md [hunk 5]
```diff
@@ -1,7 +1,10 @@
 # Changing the Default Authorization
 
-Mandatory Access Controller (MAC) and Discretionary Access Controller (DAC) use for the authorization in the Message
-Broker and it is disabled by default. User could enable it and restrict the resource access. 
+Mandatory Access Control (MAC) and Discretionary Access Control (DAC) use for the authorization in the Message
```
[the remark this answers is not in the document]
Comment (indikasampath2000) [reply docs/user/authorization-settings-guide.md]: Fixed in cc21144

File: docs/user/authorization-settings-guide.md (hunk 4 above)
Comment (indikasampath2000) [reply docs/user/authorization-settings-guide.md]: Fixed in cc21144

**Pull request merged**